src/common/preferences/preferences.jsx

```diff
@@ -0,0 +1,632 @@
+			currentHostIdx = this.state.currentProxy.hosts.length-1;
+		} else if (event.target.value == '-') {
+			this.state.currentProxy.hosts.splice(this.state.currentHostIdx, 1);
+		}
+		this.setState({currentHostIdx: currentHostIdx});
+		this.saveProxies(this.state.currentProxyIdx);
+	},
+	
+	handleHostnameChange: function(event) {
+		this.state.currentProxy.hosts[this.state.currentHostIdx] = event.target.value;
+		this.saveProxies(this.state.currentProxyIdx);
+	},
+	
+	render: function(){
+		var configuredProxies;
+		if (this.state.proxies.length) {
+			configuredProxies = this.state.proxies.map((proxy, i) => 
+				<option value={i} key={i}>{proxy.scheme}</option> 
+			);
+		} else {
+			configuredProxies = <option value={-1}></option>;
+		}
+		
+		var proxySettings = "";
+		if (this.props.transparent && this.state.currentProxy) {
+			let multiHost = this.state.currentProxy.scheme.indexOf('%h') != -1;
+			let configuredHosts = this.state.currentProxy.hosts.map((host, i) => 
+				<option value={i} key={i}>{host}</option>);
+				
+			proxySettings = (
+				<div className="group" style={{marginTop: "10px"}}>
+					<p style={{display: "flex", alignItems: "center", flexWrap: "wrap"}}>
+						<label style={{visibility: multiHost ? null : 'hidden'}}><input type="checkbox" name="autoAssociate" onChange={this.handleCheckboxChange} checked={this.state.currentProxy.autoAssociate}/>&nbsp;Automatically associate new hosts</label><br/>
+						<label><input type="checkbox" name="dotsToHyphens" onChange={this.handleCheckboxChange} checked={this.state.currentProxy.dotsToHyphens}/>&nbsp;Automatically convert hyphens to dots in proxied hostnames</label><br/>
+					</p>
+					<p style={{display: "flex", alignItems: "center"}}>
+						<label style={{alignSelf: "center", marginRight: "5px"}}>Scheme: </label>
+						<input style={{flexGrow: "1"}} type="text" name="scheme" onChange={this.handleSchemeChange} value={this.state.currentProxy.scheme}/>
+					</p>
+					<p>
+						You may use the following variables in your proxy scheme:<br/>
+						&#37;h - The hostname of the proxied site (e.g., www.example.org)<br/>
+						&#37;p - The path of the proxied page excluding the leading slash (e.g., about/index.html)<br/>
+						&#37;d - The directory path (e.g., about/)<br/>
+						&#37;f - The filename (e.g., index.html)<br/>
+						&#37;a - Any string
+					</p>
+					
+					<div style={{display: "flex", flexDirection: "column", marginTop: "10px"}}>
+						<label>Hostnames</label>
+						<select size="8" multiple onChange={this.handleHostSelectChange} value={[this.state.currentHostIdx]}>
+							{configuredHosts}
+						</select>
+						<p>
+							<input style={{minWidth: "80px", marginRight: "10px"}} type="button" onClick={this.handleHostButtonClick} value="+"/>
+							<input style={{minWidth: "80px", marginRight: "10px"}} type="button" onClick={this.handleHostButtonClick} disabled={this.state.currentHostIdx == -1} value="-"/>
```

Comment:
`Remove` instead of `-`

(The minus alone looks a little weird without a plus.)